ps_socialfollow.php

```diff
@@ -283,13 +320,27 @@ protected function updateFields()
     {
         $validator = Validation::createValidator();
         $constraints = [new Url()];
-
+        $config = [];
+        $errors = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $value = Tools::getValue('blocksocial_' . $social, '');
-            $violations = $validator->validate($value, $constraints);
-            if (0 === count($violations)) {
-                Configuration::updateValue('BLOCKSOCIAL_' . strtoupper($social), $value);
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $config[$social][$id_lang] = Tools::getValue('blocksocial_' . $social . '_' . $id_lang, '');
+                $violations = $validator->validate($config[$social][$id_lang], $constraints);
+                if (count($violations)) {
+                    $errors[] = $this->trans('Invalid URL', [], 'Admin.Notifications.Error') . ': ' . $config[$social][$id_lang];
+                }
+            }
+
+        }
+        if (empty($errors)) {
+            foreach (static::SOCIAL_NETWORKS as $social) {
+                Configuration::updateValue('BLOCKSOCIAL_' . strtoupper($social), $config[$social]);
             }
+        } else {
+            $this->validation_message = $this->displayError(implode('<br />', $errors));
```

Comment:
Nice-to-have: it's great to avoid mixing HTML with backend logic.

Could we keep the array of errors, and later transform this into a HTML statement?

We could do the data-to-HTML conversion at line 108 at the moment we send the HTML?

Reply:
I agree. Event the private property `$this->validation_message` would no longer be needed